### DISCUSSION TITLE
Why do you exclude records where 'Patv'<0  from evaluation?

### DISCUSSION THREAD

**Discussion by begovegar** [PaddlePaddle/PaddleSpatial#88] [BaiduKDDCup2022] [answered]
Hi, we don't understand why you exclude records with ['Patv'] < 0 from the evaluation metrics.
According to the paper, you should exclude records with Patv<0 (so=0) when Wdir>2.5. Both conditions should be met.
Is there anything in the report I am missing? 
Thanks,

From https://github.com/PaddlePaddle/PaddleSpatial/blob/main/apps/wpf_baseline_gru/kddcup22-sdwpf-evaluation/paddlepaddle/metrics.py:

invalid_cond = (raw_data['Patv'] < 0) | \
                   ((raw_data['Patv'] == 0) & (raw_data['Wspd'] > 2.5)) | \
                   ((raw_data['Pab1'] > 89) | (raw_data['Pab2'] > 89) | (raw_data['Pab3'] > 89)) | \
                   ((raw_data['Wdir'] < -180) | (raw_data['Wdir'] > 180) | (raw_data['Ndir'] < -720) |
                    (raw_data['Ndir'] > 720))

indices = np.where(~nan_cond & ~invalid_cond)
    prediction = pred[indices]
    targets = gt[indices]

**Comment by ramber1836** [comment, marked as answer]
Negative Patv is invalid, which should be not taken into account as well.

Regards,
Xinjiang Lu
